cmd/root_test.go

```diff
@@ -1531,6 +1531,109 @@ func TestGit(t *testing.T) {
 	)
 }
 
+func TestTreeRootExclusivity(t *testing.T) {
+	tempDir := test.TempExamples(t)
+	configPath := filepath.Join(tempDir, "/treefmt.toml")
+
+	formatterConfigs := map[string]*config.Formatter{
+		"echo": {
+			Command:  "echo", // will not generate any underlying changes in the file
+			Includes: []string{"*"},
+		},
+	}
+
+	test.ChangeWorkDir(t, tempDir)
+
+	assertExclusiveFlag := func(as *require.Assertions, err error) {
+		as.ErrorContains(err,
+			"if any flags in the group [tree-root tree-root-cmd tree-root-file] are set none of the others can be;",
+		)
+	}
+
+	assertExclusiveConfig := func(as *require.Assertions, err error) {
+		as.ErrorContains(err,
+			"only one of tree-root, tree-root-cmd or tree-root-file can be specified",
+		)
+	}
+
+	envValues := map[string][]string{
+		"tree-root":      {"TREEFMT_TREE_ROOT", "bar"},
+		"tree-root-cmd":  {"TREEFMT_TREE_ROOT_CMD", "'echo buzz'"},
```

Comment:
nit: I know this does not actually matter for this test, but this looks like a weird command.

```suggestion
			cfg.TreeRootCmd = "echo buzz"
```

(same comment throughout the test)

Reply:
I think its deliberate that the `TREEFMT_TREE_ROOT_CMD` env var is being used here, rather than directly mutating the `cfg` struct. Is it the nested quotes that you are primarily complaining about?

Reply:
As @MattSturgeon says. However it doesn't technically require the `' '` when used in an env variable, that's just a copy paste error.

Reply:
I changed the example and removed the `' '` in the env variable examples.

Reply:
Urg, sorry, I was only complaining about the nested quotes. (I moved this comment around after I wrote it.)